fn/req.go

```diff
@@ -0,0 +1,31 @@
+package fn
+
+// Req is a type to encapsulate RPC-like calls wherein we send some data
+// structure as a request, as well as a channel to receive the response on where
+// the remote goroutine will send the result.
+type Req[Input any, Output any] struct {
+	// Request is the data we are sending to the remote goroutine for
+	// processing.
+	Request Input
+
+	// Response is the channel on which we will receive the result of the
+	// remote computation.
+	Response chan<- Output
+}
+
+// NewReq is the base constructor of the Req type. It returns both the packaged
+// Req object as well as the receive side of the response channel that we will
+// listen on for the response.
+func NewReq[Input, Output any](input Input) (
```

Comment:
It's not just a shortcut because the field type of the response is a send only channel, so we return the receive-only side. This is to make the API safer. The thread that constructs the request also gets the receive side, and the request itself only has the send side. This prevents API misuse.

Reply:
Ah, I missed that part. I guess I didn't look at it close enough :see_no_evil:  That's actually pretty clever, I like it.